### PR TITLE
Implement ReplicaSet client.

### PR DESCRIPTION
**Pull request by madhusudancs**

### CODE REVIEW

File: pkg/client/unversioned/replica_sets.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+// newReplicaSets returns a ReplicaSetClient
+func newReplicaSets(c *ExtensionsClient, namespace string) *replicaSets {
+	return &replicaSets{c, namespace}
+}
+
+// List takes a selector, and returns the list of ReplicaSets that match that selector.
+func (c *replicaSets) List(opts api.ListOptions) (result *extensions.ReplicaSetList, err error) {
+	result = &extensions.ReplicaSetList{}
+	err = c.client.Get().Namespace(c.ns).Resource("replicaSets").VersionedParams(&opts, api.Scheme).Do().Into(result)
```
Comment (nikhiljindal) [inline pkg/client/unversioned/replica_sets.go]: Shouldnt the resource be "replicasets" (all lowercase)?

**Comment by nikhiljindal** [comment]
LGTM, modulo one comment.

Feel free to self apply the label after fixing that

**Comment by madhusudancs** [comment]
Made the change. Thanks for catching that. 

I am not adding the LGTM label yet. I am waiting for the previous PRs to be merged (controlling the rollout ;-) )

**Comment by madhusudancs** [comment]
Base PR #19042 merged. Adding LGTM.

**Comment by madhusudancs** [comment]
Adding LGTM back. Made some trivial comment changes based on @kargakis' comments.

**Comment by madhusudancs** [comment]
@k8s-bot test this please.

**Comment by madhusudancs** [comment]
@k8s-bot test this please.

(Tests are failing for completely unrelated reasons.)

**Comment by bgrant0607** [comment]
@k8s-oncall
This has been LGTM for over a week, has passed e2e a dozen times, and is blocking a dozen other PRs. Could we please get a manual merge since the submit queue is still broken?

**Comment by fabioy** [comment]
I just merged a number of PRs to help with test (flakiness or investigations), so let me get one cycle of green runs. I'll manually merge this one afterwards.

**Comment by fabioy** [comment]
@madhusudancs Sorry, need another rebase. I can manually merge afterwards. Thanks.

**Comment by madhusudancs** [comment]
@fabioy Rebased. Please merge.

**Comment by fabioy** [comment]
Manually merged.

**Pull request merged**